textproto/header.go

```diff
@@ -513,16 +521,25 @@ func ReadHeader(r *bufio.Reader) (Header, error) {
 		maxLines, kv, err = readContinuedLineSlice(r, maxLines)
 		if len(kv) == 0 {
 			if err == io.EOF {
-				err = nil
+				//err = nil
+				return newHeader(fs), finalErr
 			}
-			return newHeader(fs), err
+			if finalErr == nil && err != nil {
+				finalErr = err
+			} else {
+				if err != nil {
+					finalErr = fmt.Errorf("%w: %v", finalErr, err)
```

Comment:
This seems to construct a chain of errors for each failed to parse line.
Why is it necessary? Is there a need to know the amount or content of malformed fields? Why this is not done by collecting these into an error structure?

Reply:
an error structure needs to be filled as slice, formated etc. its alot of more code. But the function ReadHeader() just returns one error, with the new error system this one error can contain more types, not just one. I think its the new way to handle it.

The downside is that go.mod needs to be changed to go 1.13

Reply:
The question remains. How important it is to preserve all lines that failed parsing? What would user code do with it?

Also I sense the possibility of DoS - untrusted source could input a large malformed string as header, causing your code to build a large error chain, possibly consuming up to 10 times the size of input. Can we be sure this is not a concern? 

Logging such error as is is not safe either as it includes verbatim malformed line, creating the possibility of logs poisoning.

Reply:
ok thats a good point. So just return the last error?

Reply:
I believe this is the most appropriate way.